pkg/controllers/cluster/cluster_controller.go

```diff
@@ -654,22 +654,30 @@ func (c *Controller) collectIDForClusterObjectIfNeeded(ctx context.Context) (err
 
 		clusterClient, err := util.NewClusterClientSet(cluster.Name, c.Client, nil)
 		if err != nil {
-			errs = append(errs, err)
+			errs = appendErr(errs, err, cluster.Name)
 			continue
 		}
 
 		id, err := util.ObtainClusterID(clusterClient.KubeClient)
 		if err != nil {
-			errs = append(errs, err)
+			errs = appendErr(errs, err, cluster.Name)
 			continue
 		}
 		cluster.Spec.ID = id
 
 		err = c.Client.Update(ctx, cluster)
 		if err != nil {
-			errs = append(errs, err)
+			errs = appendErr(errs, err, cluster.Name)
 			continue
 		}
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+// appendErr is a helper function to collect cluster-specific errors.
+func appendErr(errs []error, err error, cluster string) []error {
+	if err != nil {
```

Comment:
It seems this judgment is redundant, furthermore, `appendErr` is unnecessary.

Reply:
ok, let me simplify it.